lib/types/types.go

```diff
@@ -95,8 +95,12 @@ func (d Duration) String() string {
 }
 
 // ParseExtendedDuration is a helper function that allows for string duration
-// values containing days.
+// values containing days. If no units are provided, milliseconds are assumed.
```

Comment:
This also applies to other configurations such as:
- setup/teardownTimeout
- gracefulStop/rampdown
- timeunit
- duration
- starttime
... more 
I am not saying this is bad idea, I am just pointing out this is breaking change even if it's likely a welcome one
Also ... as far as I can see ... this is not used consistantly so some options will be unmarshalled from JSON to this but not from the cli flags for example `min-iteration-duration` will use time.ParseDuration as this is [what pflag says](https://github.com/spf13/pflag/blob/master/duration.go#L15-L18) or maybe .. not, not certain on whether there is somethign that will check for `UnmarshalText` before that. 

But I would argue this should've been another function in this PR and have this merging be another PR in which this is more tested and looked into
cc @na--

Reply:
hmm I think it's not going to be a big issue and we have to address https://github.com/loadimpact/k6/issues/1305 eventually... @MStoykov keep in mind that this won't actually be used for integer JS/JSON values because of https://github.com/loadimpact/k6/blob/caf82ab98ec3b5dc581519a5269c276ca05a2306/lib/types/types.go#L154-L160
So I'm not sure how much effect this will actually have

Reply:
but yeah, maybe we shouldn't make half-measures... so I'm fine if we use a separate function this time, with a `TODO` to switch completely to it and address #1305 in another PR

Reply:
Looking in the code that you just linked ... I am even more of the opinion that this should be a separate functio now ... as this seems to make things ... worse :D

Reply:
fair enough - I can't think of an example how to break it, but it'd definitely be cleaner, so :+1: from me

Reply:
Hey, thanks for the heads up, I agree that it's safer to use a different function. I changed it in the recent push, along with some squashing since I had to fix the merge conflicts anyway.